### DISCUSSION TITLE
Workflow doesn't trigger on some branches

### DISCUSSION THREAD

**Discussion by rgov** [community/community#27018] [Actions]
<p>I have <a href="https://github.com/WHOIGit/ndsf-sealog-server" rel="nofollow noopener">a repository</a> and <a href="https://github.com/WHOIGit/ndsf-sealog-server/blob/master/.github/workflows/docker.yml" rel="nofollow noopener">workflow</a> that is configured to do a build whenever there is a push to any branch:</p>
<pre><code>on: [push, pull_request]
</code></pre>
<p>And this works fine whenever I make a push to master or a feature branch.</p>
<p>I also have branches prefixed with the string upstream/. I have a local system with a cron job to pull branches from an upstream repo, add this prefix, and then push them up to GitHub.</p>
<p>However, the CI workflow never triggers for these pushes. Why is that? Perhaps workflows do not trigger when a push happens using a read/write deploy credential instead of a user credential?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>There is no .github/workflow/ directory on the upstream branches. Workflows for push or pull_request are read from the affected branch, and there aren’t any workflow definitions in the upstream branches.</p>
